sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_visits_daily_v2/query.sql

```diff
@@ -0,0 +1,302 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_info.client_id AS client_id,
+    SAFE_CAST(
+      mozfun.norm.browser_version_info(client_info.app_display_version).major_version AS INT64
+    ) AS app_version,
+    normalized_os AS os,
+    normalized_channel AS channel,
+    client_info.locale AS locale,
+    client_info.app_display_version AS browser_version,
+    normalized_country_code AS country,
+    metrics.string.newtab_homepage_category AS newtab_homepage_category,
+    metrics.string.newtab_newtab_category AS newtab_newtab_category,
+    metrics.boolean.pocket_enabled AS organic_content_enabled,
+    metrics.boolean.pocket_sponsored_stories_enabled AS sponsored_content_enabled,
+    metrics.boolean.topsites_sponsored_enabled AS sponsored_topsites_enabled,
+    metrics.boolean.topsites_enabled AS organic_topsites_enabled,
+    metrics.boolean.newtab_search_enabled AS newtab_search_enabled,
+    category AS event_category,
+    name AS event_name,
+    extra AS event_details,
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    -- selecting the events we need to compute on
+    -- Do we need to include any other events for computing "all_visits"?
+    AND (
+      category = 'newtab'
+      AND name IN ('opened')
+      OR category = 'newtab.search'
+      AND name IN ('issued')
+      OR category = 'newtab.search.ad'
+      AND name IN ('click')
+      OR category = 'pocket'
+      AND name IN (
+        'click',
+        'impression',
+        'dismiss',
+        'thumb_voting_interaction',
+        'thumb_voting_interaction'
+      )
+      OR category = 'topsites'
+      AND name IN ('click', 'impression', 'dismiss')
+      OR category = 'newtab'
+      AND name IN (
+        'weather_change_display',
+        'weather_open_url_provider',
+        'weather_location_selected'
+      )
+      OR category = 'newtab'
+      AND name IN (
+        'wallpaper_click',
+        'wallpaper_category_click',
+        'wallpaper_highlight_cta_click',
+        'wallpaper_highlight_dismissed'
+      )
+      OR category = 'newtab'
+      AND name IN (
+        'topic_selection_open',
+        'topic_selection_dismiss',
+        'topic_selection_topics_saved'
+      )
+      OR category = 'newtab'
+      AND name IN (
+        'sections_block_section',
+        'sections_follow_section',
+        'sections_unblock_section',
+        'sections_unfollow_section'
+      )
+      OR category = 'newtab'
+      AND name IN ('inline_selection_click')
+    )
```

Comment:
Parentheses need to be added to group the `category` and `name` checks for each category, at least for the sake of readability.  In the existing `newtab_visits` the filtering only happens by category (`category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad', 'pocket')`).  This may be preferable, it's easier to read and more robust to the additional of new events.

Reply:
The reason I applied the event names for each category because some of the event names are overloaded (example: `click`). By specifying events with category can give better control over the definitions.

A question related to visit count `newtab_visit_count` in the existing `newtab_clients`: does the `newtab_visit_count` expected to could visits associated with all events in these categories `category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad', 'pocket')` Or would certain events be excluded in the computation?

Reply:
I can't think of a scenario where we could want to exclude a specific event from a category.

Reply:
> I can't think of a scenario where we could want to exclude a specific event from a category.

I could use some guidance in this:
My thought was that `newtab` is a general category is used for several (perhaps unrelated) events. Therefore, setting category = 'newtab' could potentially include those events that we may not need.
If this is not a concern (for now as well as for the future), then we could perhaps exclude the event _name_ filter in the initial cte. It would make the query simpler for sure.

Reply:
So there unfortunately doesn't seem to be an easy way to just look at the "names" associated with a "category".  Best I can tell is that the categories are the top-level keys of this while, and the names for that categories are the keys within it (for us, I think we only care about the keys with `type:event`)

https://github.com/mozilla-firefox/firefox/blob/3bda5e2fd3dcc60d0bf50cafed641a43bfba0ec2/browser/components/newtab/metrics.yaml

Reply:
Looking through it there definitely seems to be events we can ignore, (IE fakespot associated ones), so I think the name-level filter might make sense.  In the query, could you group all the names associated with a single category together to make it easier to check?